src/components/InstanceFilters/InstanceFilters.js

```diff
@@ -185,6 +186,10 @@ const InstanceFilters = props => {
     props.data
   );
 
+  const isStaffSuppressFilterAvailable = stripes.hasPerm('ui-inventory.instance.view-staff-suppressed-records');
+
+  useStaffSuppressInitialValue(onChange, props.data.query);
```

Comment:
Let's call the `onChange` callback with both the selected option and the staff suppress. This will allow us to make only one change in history and have only one request with the same endpoint to the server.

Reply:
`onFilterChangeHandler` already merges staff suppress and other selected filters, so we don't have to do that anywhere else.
![image](https://github.com/folio-org/ui-inventory/assets/19309423/cd864937-3f27-48a0-9f1b-8991f68ea8d7)

The first request is made by default when the app is mounted, and query string is built in `buildQuery` function
In that function we don't know if this is the very first request and we don't have staff suppress filter because it's not in the url, or we don't have it because a user has unchecked the option in the facet component.
We can only safely change filters in the function if search is empty or the user doesn't have the permission so we know for sure that adding staff suppress value if it's not there does not go agains user's actions

Reply:
I'm referring having two instance requests and two facets requests after selecting the facet option. 

https://github.com/folio-org/ui-inventory/assets/77053927/14194f25-723f-48d8-b949-4e87d9f206c8

Reply:
It is working now, thanks. 

I also think that we need to pass location into the hook instead of using it inside, since this hook will be used in the `find-instance` plugin and extracted to `stripes-inventory-componets`. There is no plugin's URL. 

It would be nice to change `history` via `replace` instead of `push` when we change filters automatically.